app/uk/gov/hmrc/pdsauthcheckerapi/controllers/AuthorisationController.scala

```diff
@@ -32,13 +37,13 @@ class AuthorisationController @Inject()(cc: ControllerComponents, config: Config
     "message" -> "Auth Type provided is not supported"
   ))
 
-  def authorise(authType: String): Action[AnyContent] = Action { _ =>
-    if (!supportedAuthTypes.contains(authType)) {
-      InvalidAuthTypeResponse
+  def authorise: Action[PdsAuthRequest] = Action(parse.json[models.PdsAuthRequest]).async { implicit request =>
+    val pdsAuthRequestBody = request.body
+    if (!supportedAuthTypes.contains(pdsAuthRequestBody.authType)) {
+      Future.successful(InvalidAuthTypeResponse)
     } else {
-      authType match {
-        case "UKIM" => NoContent
-        case _ => InvalidAuthTypeResponse
+      pdsService.getValidatedCustoms(pdsAuthRequestBody).map { pdsAuthResponse =>
+        Ok(Json.toJson(pdsAuthResponse))
```

Comment:
Nothing here covers 400-500 responses, it assumes the happy path, we need to be able to control our responses here.

Reply:
This should now be covered by Play handling any internal exceptions, and the parse.Json handles bad requests, and the connector now handling non-200 responses from PDS. Happy to have a chat if this doesn't meet expectations.

Reply:
I'm sure I'll think of a reason to change it later, but looks good for now